src/hoc/RealtimeTemplates.tsx

```diff
@@ -28,37 +28,30 @@ export const RealTimeTemplates = ({
   const { templates } = useSelector(selectCreateTemplate)
   const { token } = useSelector(selectTaskBoard)
 
-  const userId = tokenPayload?.internalUserId || tokenPayload?.clientId
-  const userRole = tokenPayload?.internalUserId
-    ? AssigneeType.internalUser
-    : tokenPayload?.clientId
-      ? AssigneeType.client
-      : undefined
+  const userId = tokenPayload?.internalUserId
+  const userRole = AssigneeType.internalUser
 
   if (!userId || !userRole) {
-    console.error(`Failed to authenticate a realtime connection for id ${userId} with role ${userRole}`)
+    console.error(`Failed to authenticate a realtime connection for id ${userId} with role ${userRole}`) //client users cannot connect to the templates real time channel
   }
 
   const handleTemplatesRealTimeUpdates = (payload: RealtimePostgresChangesPayload<RealTimeTemplateResponse>) => {
     if (payload.eventType === 'INSERT') {
       let canUserAccessTask = payload.new.workspaceId === tokenPayload?.workspaceId
-      if (userRole === AssigneeType.client) {
-        canUserAccessTask = false
-      }
       if (canUserAccessTask) {
         templates
           ? store.dispatch(
-              setTemplates([...templates, { ...payload.new, createdAt: new Date(payload.new.createdAt + 'Z') }]),
+              setTemplates([{ ...payload.new, updatedAt: new Date(payload.new.updatedAt + 'Z') }, ...templates]),
             )
-          : store.dispatch(setTemplates([{ ...payload.new, createdAt: new Date(payload.new.createdAt + 'Z') }]))
+          : store.dispatch(setTemplates([{ ...payload.new, updatedAt: new Date(payload.new.updatedAt + 'Z') }]))
```

Comment:
Let's move this to a dedicated util for reuse. Something that takes in the timestamp, validates it to not have 'Z' and appends it if necessary.